src/cmap/connection_pool.ts

```diff
@@ -437,7 +525,31 @@ function createConnection(pool: ConnectionPool, callback?: Callback<Connection>)
       connection.on(event, (e: any) => pool.emit(event, e));
     }
 
-    pool.emit(ConnectionPool.CONNECTION_POOL_CREATED, new ConnectionCreatedEvent(pool, connection));
+    pool.emit(ConnectionPool.CONNECTION_CREATED, new ConnectionCreatedEvent(pool, connection));
+
+    if (pool.loadBalanced) {
+      const onPin = (pinType: string) => {
```

Comment:
Seems like these could be inlined? is there a benefit to pulling them out into variables?

Reply:
No benefit, just my preferred style. I've inlined.

Reply:
Inlining them means you can remove the type annotation because the static `'pinned'` coerces the type of the second argument.

Reply:
Removed.